program/src/processor.rs

```diff
@@ -365,6 +372,11 @@ fn process_initialize_escrow(program_id: &Pubkey, accounts: &[AccountInfo]) -> P
         return Err(PaladinLockupError::IncorrectEscrowTokenAccount.into());
     }
 
+    // Jon: this won't work for mint with extensions, since we'll need to allocate more
+    // space. If we want to keep it simple, we could by convention require the ATA for
+    // the escrow authority at that mint, that way we know the account isn't tampered
+    // with, and we can remove this instruction. Or we can copy the logic in the ATA
+    // program. I lean towards the first.
```

Comment:
Yeah let's just enforce it's the ATA. And is there a reason to initialize the escrow PDA? There's no data stored on it, so we can just keep it as an uninitialized signer

Reply:
Sorry, all I mean by "initialize" is "assign", since the instruction has to `invoke_signed` to assign it to itself, which I chalked up as effectively _initializing_ the PDA signer.

Reply:
Right, we definitely need to provide the account, but it doesn't need to be assigned to the program

Reply:
Ohhh interesting. For some reason I thought that would invalidate the PDA signature. Nice!